fastparquet/core.py

```diff
@@ -150,8 +151,12 @@ def read_col(column, schema_helper, infile, use_cat=False):
     use_cat: bool (False)
         If this column is encoded throughout with dict encoding, give back
         a pandas categorical column; otherwise, decode to values
+    grab_dict: bool (False)
+        Short-cut mode to return the dictionary values only - skips the actual
+        data.
```

Comment:
I know... I can't think of a way to uncouple it, because when reading a column chunk, you don't know whether it will have a dictionary, so you must read the page header - but if it's not a dictionary, that then becomes the page header needed later.